lib/extty.ex

```diff
@@ -96,6 +96,13 @@ defmodule ExTTY do
     {:noreply, %{state | buf: new_buf}}
   end
 
+  @impl GenServer
+  def terminate(reason, state) do
+    # The `IEx.Server` and `IEx.Evaluator` don't go away with a simple `:normal` exit.
+    Process.exit(state.group, :kill)
```

Comment:
The problem with this is that everything is linked correctly except for `IE.Evaluator` which aggressively monitors and doesn't close itself. So this gets past that, but there is potential that the group gets closed before we get to this terminate callback _or_ the terminate callback doesn't get called.

But after some deep dives, it might be the simplest path forward. The other option is to write our own `IEx.Evaluator` that does less aggressive things and then supply that as an `evaluator: ExTTY.ElixirEvaluator.start()` option when spawning the shell